### PR TITLE
Fixed #978: Fixes several issues for the feature attribute view when handling extended configurations

### LINKED ISSUE #978: Feature Attributes View not working as intended for Extended Configurations
#### Prerequisitives
* **Eclipse Version:**  `4.12`
* **FeatureIDE-Version:**  `Release3.6 Branch`
* **Operating system:** `Windows 10`

### Issue description
grafik
This is always shown when using the feature attribute view while working with an extended configuration.
Expected Behavior:
The attributes of some/all features with the "Configurable"-flag are shown. 
Viable options to decide which features to display I can think of:
(A) all features
(B) all features included in the configuration
(C) only the feature marked by the last click

### PR DESCRIPTION
**Pull request by Subaro**

### CODE REVIEW

File: plugins/de.ovgu.featureide.fm.attributes/src/de/ovgu/featureide/fm/attributes/base/impl/BooleanFeatureAttribute.java [hunk 1]
```diff
@@ -81,4 +81,9 @@ public IFeatureAttribute cloneAtt(IFeature feature) {
 	public IFeatureAttribute cloneRecursive(IFeature feature) {
 		return new BooleanFeatureAttribute(feature, this.getName(), this.getUnit(), null, this.isRecursive(), this.isConfigurable());
 	}
+
+	@Override
+	public boolean isValidValue(String value) {
+		return true;
```
Comment (PaulWestphal) [inline plugins/de.ovgu.featureide.fm.attributes/src/de/ovgu/featureide/fm/attributes/base/impl/BooleanFeatureAttribute.java]: is every string a valid value?

File: plugins/de.ovgu.featureide.fm.attributes/src/de/ovgu/featureide/fm/attributes/base/impl/StringFeatureAttribute.java [hunk 2]
```diff
@@ -86,4 +86,9 @@ public IFeatureAttribute cloneAtt(IFeature feature) {
 	public IFeatureAttribute cloneRecursive(IFeature feature) {
 		return new StringFeatureAttribute(feature, this.getName(), this.getUnit(), null, this.isRecursive(), this.isConfigurable());
 	}
+
+	@Override
+	public boolean isValidValue(String value) {
+		return true;
```
Comment (PaulWestphal) [inline plugins/de.ovgu.featureide.fm.attributes/src/de/ovgu/featureide/fm/attributes/base/impl/StringFeatureAttribute.java]: same for this

File: plugins/de.ovgu.featureide.fm.attributes/src/de/ovgu/featureide/fm/attributes/base/impl/BooleanFeatureAttribute.java (hunk 1 above)
Comment (SundermannC) [reply plugins/de.ovgu.featureide.fm.attributes/src/de/ovgu/featureide/fm/attributes/base/impl/BooleanFeatureAttribute.java]: This is intended, but open for discussion. Currently, anything resolves to false except capitalizations of true .

File: plugins/de.ovgu.featureide.fm.attributes/src/de/ovgu/featureide/fm/attributes/base/impl/StringFeatureAttribute.java (hunk 2 above)
Comment (SundermannC) [reply plugins/de.ovgu.featureide.fm.attributes/src/de/ovgu/featureide/fm/attributes/base/impl/StringFeatureAttribute.java]: Currently intended. We can discuss possible limitations though (e.g., regarding length).

File: plugins/de.ovgu.featureide.fm.attributes/src/de/ovgu/featureide/fm/attributes/view/listener/FeatureAttributeSelectionListener.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+import org.eclipse.jface.viewers.IStructuredSelection;
+import org.eclipse.jface.viewers.SelectionChangedEvent;
+import org.eclipse.jface.viewers.TreeViewer;
+
+import de.ovgu.featureide.fm.attributes.base.impl.FeatureAttribute;
+import de.ovgu.featureide.fm.attributes.view.FeatureAttributeView;
+import de.ovgu.featureide.fm.core.base.IFeature;
+import de.ovgu.featureide.fm.ui.editors.featuremodel.editparts.FeatureEditPart;
+
+/***
```
Comment (PaulWestphal) [inline plugins/de.ovgu.featureide.fm.attributes/src/de/ovgu/featureide/fm/attributes/view/listener/FeatureAttributeSelectionListener.java]: too many *

File: plugins/de.ovgu.featureide.fm.core/src/de/ovgu/featureide/fm/core/base/event/FeatureIDEEvent.java [hunk 4]
```diff
@@ -37,151 +37,159 @@
...
-		 * each feature model would have to search the explanation for the relevant reason again. As such, each event of this type carries the respective reason
-		 * so the feature model element does not have to look for it itself.
-		 */
-		ACTIVE_REASON_CHANGED,
-		/**
-		 * Default. Do nothing.
-		 */
-		// TODO !!! implement feature order event
-		DEFAULT, FEATURE_ORDER_CHANGED,
+	FEATURE_MODEL_OPERATION,
```
Comment (PaulWestphal) [inline plugins/de.ovgu.featureide.fm.core/src/de/ovgu/featureide/fm/core/base/event/FeatureIDEEvent.java]: this one doesn't have a description

File: plugins/de.ovgu.featureide.fm.core/src/de/ovgu/featureide/fm/core/base/event/FeatureIDEEvent.java [hunk 5]
```diff
@@ -37,151 +37,159 @@
...
+	 * The active reason changed. Events of this type are fired for feature model elements when the active explanation has changed. It would be possible to
+	 * instead simply notify each affected feature model element of the new active explanation. However, this would lead to a negative performance impact as
+	 * each feature model would have to search the explanation for the relevant reason again. As such, each event of this type carries the respective reason so
+	 * the feature model element does not have to look for it itself.
+	 */
+	ACTIVE_REASON_CHANGED,
+	/**
+	 * Default. Do nothing.
+	 */
+	// TODO !!! implement feature order event
```
Comment (PaulWestphal) [inline plugins/de.ovgu.featureide.fm.core/src/de/ovgu/featureide/fm/core/base/event/FeatureIDEEvent.java]: make this TODO: so it's properly flagged

**Pull request merged**